staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/stdlib.go

```diff
@@ -0,0 +1,585 @@
+//
+// Examples:
+//
+//     [1, 3].sum() // returns 4
+//     [].sum() // returns 0
+//
+//
+// min / max
+//
+// Returns the minimum/maximum valued element of the provided list. Supports all comparable types.
+// If the list is empty, an error is returned.
+//
+//     <list<T>>.min() <T>, T must be a comparable type
+//     <list<T>>.max() <T>, T must be a comparable type
+//
+// Examples:
+//
+//     [1, 3].min() // returns 1
+//     [1, 3].max() // returns 3
+//     [].min() // error
+//
+//
+// indexOf / lastIndexOf
+//
+// Returns the positional index of the provided element in the list. Supports all equatable types.
+// If the element is not found, -1 is returned.
+//
+//     <list<T>>.indexOf(<T>) <int>
+//     <list<T>>.lastIndexOf(<T>) <int>
+//
+// Examples:
+//
+//     [1, 2, 2, 3].indexOf(2) // returns 1
+//     [1, 2, 2, 3].lastIndexOf(2) // returns 2
+//     [1].indexOf(2) // returns -1
+//
+//
+// find / findAll
+//
+// Returns substrings that match the provided regular expression. find returns the first match. findAll may optionally
+// be provided a limit. If the limit is set and >= 0, no more than the limit number of matches are returned.
+//
+//     <string>.find(<string>) <string>
+//     <string>.findAll(<string>) <list <string>>
+//     <string>.findAll(<string>, <int>) <list <string>>
+//
+// Examples:
+//
+//     "abc 123".find('[0-9]*') // returns '123'
+//     "abc 123".find('xyz') // returns ''
+//     "123 abc 456".findAll('[0-9]*') // returns ['123', '456']
+//     "123 abc 456".findAll('[0-9]*', 1) // returns ['123']
+//     "123 abc 456".findAll('xyz') // returns []
+//
+//
+// url
+//
+// Converts a string to a URL or results in an error if the string is not a valid URL.
+//
+//     url(<string>) <URL>
+//
+//
+// isURL
+//
+// Returns true if a string is a valid URL.
+//
+//     <string>.isURL() <bool>
+//
+//
+// getScheme / getUsername / getPassword / getHostname / getPort / getPath / getQuery / getFragment
+//
+// Return the parsed components of the URL.
+//
+//     <URL>.getScheme() <string>
+//     <URL>.getUsername() <string>
+//     <URL>.getPassword() <string>
+//     <URL>.getHostname() <string>
+//     <URL>.getPort() <string>
+//     <URL>.getPath() <string>
+//     <URL>.getQuery() <map <string>, <list <string>>
+//     <URL>.getFragment() <string>
+//
+// Examples:
+//     url('https://example.com/').getScheme() // returns 'https'
+//     url('https://user:pass@example.com/').getUsername() // returns 'user'
+//     url('https://user:pass@example.com/').getPassword() // returns 'pass'
+//     url('https://example.com:80/path').getHostname() // returns 'example.com'
+//     url('https://127.0.0.1:80/path').getHostname() // returns '127.0.0.1'
+//     url('https://[::1]:80/path').getHostname() // returns '::1'
+//     url('https://example.com:80/path').getPort() // returns '80'
+//     url('https://example.com/path').getPath() // returns '/path'
```

Comment:
give an example of getPath() returning a %-escaped path

`url('https://example.com/path with spaces/').getPath()` returns `/path%20with%20spaces/`, I think